### PR TITLE
Compile Crypto FFI for MacOS

### PR DESCRIPTION
**Pull request by Anderas**
- Add steps to compile Crypto FFI for macOS
- Upgrade uniff-rs to include swift [codegen bug fix](https://github.com/mozilla/uniffi-rs/pull/1365)
- Update Podspec version to release new CocoaPods version

### CODE REVIEW

File: bindings/matrix-sdk-crypto-ffi/src/lib.rs [hunk 1]
```diff
@@ -471,14 +473,6 @@ mod uniffi_types {
     pub use crate::{backup_recovery_key::BackupRecoveryKey, machine::OlmMachine};
 }
 
-#[allow(warnings)]
-mod generated {
```
Comment (jplatte) [inline bindings/matrix-sdk-crypto-ffi/src/lib.rs:475]: Why was this renamed / moved into its own file?

File: bindings/apple/build_crypto_xcframework.sh [hunk 2]
```diff
@@ -10,7 +10,7 @@ TARGET_DIR="${SRC_ROOT}/target"
 
 GENERATED_DIR="${SRC_ROOT}/generated"
 if [ -d "${GENERATED_DIR}" ]; then rm -rf "${GENERATED_DIR}"; fi
-mkdir -p ${GENERATED_DIR}
+mkdir -p "${GENERATED_DIR}/macos" "${GENERATED_DIR}/simulator"
```
Comment (jplatte) [inline bindings/apple/build_crypto_xcframework.sh:13]:
```suggestion
mkdir -p "${GENERATED_DIR}/{macos,simulator}"
```

File: bindings/matrix-sdk-crypto-ffi/src/lib.rs (hunk 1 above)
Comment (Anderas) [reply bindings/matrix-sdk-crypto-ffi/src/lib.rs:475]: This is to use the same uniffi setup as `matrix-rust-sdk` as well as [official guide](https://mozilla.github.io/uniffi-rs/tutorial/Rust_scaffolding.html) when we do not seem to be using custom location.

**Comment by jplatte** [comment]
Okay I think we have other improvements planned in this area, but apart from the UniFFI thing which I wouldn't say is clearly better (but also not clearly worse), this is strictly better than what we have right now.

**Review by jplatte — APPROVED** [review]

**Comment by jplatte** [comment]
CI failures are unrelated and should not happen again after merge.
